### PR TITLE
UPDATE Draw Card Use Case

### PR DESCRIPTION
**Pull request by mariamustard21**
Unfinished pull request for DrawCard!

### CODE REVIEW

**Comment by no7thing** [comment]
Good progress so far, make sure you implement and fix the input data file, as it is important! otherwise good job!

**Review by Maxlyu254 — CHANGES_REQUESTED** [review]
You shouldn't merge this to the main branch yet, the program failed to pass the java classroom check. Try to find out what's wrong.

**Review by Maxlyu254 — APPROVED** [review]
Although there are some problems, it can compile, so pull request approved

**Review by celinayueh — DISMISSED** [review]
Good job finishing the use case and following clean architecture. Some changes will need to be made to follow MVVM model.

**Review by sinemsoylemezoglu — DISMISSED** [review]
Good job on the Draw Card use case Maria. Need to change the Presenter class.

**Review by Maxlyu254 — CHANGES_REQUESTED** [review]
Your classroom checks failed, please try to fix that problem

**Comment by zhenzeyang** [comment]
You made a lot progress! Good job! For now you need to make your code pass the checker and update the view model

File: src/main/java/DrawCard/DrawCardInteractor.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import Cards.Card;
+import Cards.GainCashCard;
+import Cards.LoseCashCard;
+import MainEntities.Deck;
+import MainEntities.Player;
+
+public class DrawCardInteractor implements DrawCardInputBoundary {
+
+    final DrawCardOutputBoundary output;
+    final Player player;
```
Comment (adelinewang3) [inline src/main/java/DrawCard/DrawCardInteractor.java:12]: you should access the player information from the CampaignAccess

File: src/main/java/DrawCard/DrawCardInputData.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+     * @param playerName the name of the player who has to draw a Card.
+     */
+    private Deck deckType;
+
+    /**
+     *
+     * @param deckType The type of Deck that the player has to draw on.
+     */
+
+    public DrawCardInputData(String playerName, Deck deckType) {
```
Comment (adelinewang3) [inline src/main/java/DrawCard/DrawCardInputData.java:20]: Your use case does not need any input data. Once a player lands on a tile, the advance use case will determine what tile it is and call the appropriate use case (in jail, draw card, pay rent, etc.).

**Review by adelinewang3 — COMMENTED** [review]
My use case doesn't use input data as well, you can take a look at it to see how you might implement yours

**Pull request closed without merging**